build/linux/debian/dependencies-generator.ts

```diff
@@ -0,0 +1,149 @@
+// The reference dependencies, which one has to update when the new dependencies
+// are valid, are in dep-lists.ts
+const FAIL_BUILD_FOR_NEW_DEPENDENCIES: boolean = true;
+
+export function getDependencies(buildDir: string, applicationName: string, arch: ArchString, sysroot: string): string[] {
+	// Get the files for which we want to find dependencies.
+	const nativeModulesPath = path.join(buildDir, 'resources', 'app', 'node_modules.asar.unpacked');
+	const findResult = spawnSync('find', [nativeModulesPath, '-name', '*.node']);
+	if (findResult.status) {
+		console.error('Error finding files:');
+		console.error(findResult.stderr.toString());
+		return [];
+	}
+
+	const files = findResult.stdout.toString().trimEnd().split('\n');
+
+	const appPath = path.join(buildDir, applicationName);
+	files.push(appPath);
+
+	// Add chrome sandbox and crashpad handler.
+	files.push(path.join(buildDir, 'chrome-sandbox'));
+	files.push(path.join(buildDir, 'chrome_crashpad_handler'));
+
+	// Generate the dependencies.
+	const dependencies: Set<string>[] = files.map((file) => calculatePackageDeps(file, arch, sysroot));
+	// Add additional dependencies.
+	const additionalDepsSet = new Set(additionalDeps);
+	dependencies.push(additionalDepsSet);
+
+	// Merge all the dependencies.
+	const mergedDependencies = mergePackageDeps(dependencies);
+	let sortedDependencies: string[] = [];
+	for (const dependency of mergedDependencies) {
+		sortedDependencies.push(dependency);
+	}
+	sortedDependencies.sort();
+
+	// Exclude bundled dependencies
+	sortedDependencies = sortedDependencies.filter(dependency => {
+		return !bundledDeps.some(bundledDep => dependency.startsWith(bundledDep));
+	});
+
+	const referenceGeneratedDeps = referenceGeneratedDepsByArch[arch];
+	if (JSON.stringify(sortedDependencies) !== JSON.stringify(referenceGeneratedDeps)) {
+		const failMessage = 'The dependencies list has changed.'
+			+ '\nOld:\n' + referenceGeneratedDeps.join('\n')
+			+ '\nNew:\n' + sortedDependencies.join('\n');
+		if (FAIL_BUILD_FOR_NEW_DEPENDENCIES) {
+			throw new Error(failMessage);
+		} else {
+			console.warn(failMessage);
+		}
+	}
+
+	return sortedDependencies;
+}
+
+// Based on https://source.chromium.org/chromium/chromium/src/+/main:chrome/installer/linux/debian/calculate_package_deps.py.
+function calculatePackageDeps(binaryPath: string, arch: ArchString, sysroot: string): Set<string> {
+	try {
+		if (!(statSync(binaryPath).mode & constants.S_IXUSR)) {
+			throw new Error(`Binary ${binaryPath} needs to have an executable bit set.`);
+		}
+	} catch (e) {
+		// The package might not exist. Don't re-throw the error here.
+		console.error('Tried to stat ' + binaryPath + ' but failed.');
+	}
+
+	// Do a sparse checkout to get the Chromium dpkg-shlibdeps file.
+	const tmp = tmpdir();
+	const chmodResult = spawnSync('chmod', ['+x', `${__dirname}/checkout-shlibdeps.sh`]);
+	if (chmodResult.status !== 0) {
+		throw new Error('Cannot set chmod +x: ' + chmodResult.stderr);
+	}
```

Comment:
can you change the executable permission of the file locally and check it in, git will preserve the permissions and this step will not be required.